include/dsn/dist/distributed_lock_service.h

```diff
@@ -69,7 +69,7 @@ namespace dsn
             /*
              * initialization routine
              */
-            virtual error_code initialize() = 0;
+            virtual error_code initialize(const char* work_dir, const char* lock_root) = 0;
```

Comment:
Quite the opposite, I think this way can reduce coupling between caller and implementation. It we just pass a general string in parameter, and leave the implementation to interpret it in it's own way, then we will have to look into the implementation before use the service, which will break the transparency of interface. On the other hand, it may also be difficult to change to another implementation if it interprets the parameter in another way. So I think it's better to pass all possible useful parameters explicitly, and leave implementations to use them as they need, even though some parameters may be not used. What do you think, @imzhenyu ?